test/xUnit/csharp/test_Prediction.cs

```diff
@@ -0,0 +1,190 @@
+                results = CommandPrediction.PredictInput(ast, tokens, millisecondsTimeout: 1000).Result;
+                Assert.Single(results);
+
+                PredictionResult res = results[0];
+                Assert.Equal(fast.Id, res.Id);
+                Assert.Equal(2, res.Suggestions.Count);
+                Assert.Equal($"Hello world TEST-1 from {fast.Name}", res.Suggestions[0].SuggestionText);
+                Assert.Equal($"Hello world TeSt-2 from {fast.Name}", res.Suggestions[1].SuggestionText);
+
+                // Expect the results from both 'slow' and 'fast' predictors
+                // Same here -- the specified timeout is exaggerated to make the test reliable.
+                // xUnit must spin up a lot tasks, which makes the test unreliable when the time difference between 'delay' and 'timeout' is small.
+                results = CommandPrediction.PredictInput(ast, tokens, millisecondsTimeout: 4000).Result;
+                Assert.Equal(2, results.Count);
+
+                PredictionResult res1 = results[0];
+                Assert.Equal(slow.Id, res1.Id);
+                Assert.Equal(2, res1.Suggestions.Count);
+                Assert.Equal($"Hello world TEST-1 from {slow.Name}", res1.Suggestions[0].SuggestionText);
+                Assert.Equal($"Hello world TeSt-2 from {slow.Name}", res1.Suggestions[1].SuggestionText);
+
+                PredictionResult res2 = results[1];
+                Assert.Equal(fast.Id, res2.Id);
+                Assert.Equal(2, res2.Suggestions.Count);
+                Assert.Equal($"Hello world TEST-1 from {fast.Name}", res2.Suggestions[0].SuggestionText);
+                Assert.Equal($"Hello world TeSt-2 from {fast.Name}", res2.Suggestions[1].SuggestionText);
+            }
+            finally
+            {
+                SubsystemManager.UnregisterSubsystem<ICommandPredictor>(slow.Id);
+                SubsystemManager.UnregisterSubsystem(SubsystemKind.CommandPredictor, fast.Id);
+            }
+        }
+
+        [Fact]
+        public static void Feedback()
+        {
+            MyPredictor slow = MyPredictor.SlowPredictor;
+            MyPredictor fast = MyPredictor.FastPredictor;
+
+            try
+            {
+                // Register 2 predictor implementations
+                SubsystemManager.RegisterSubsystem<ICommandPredictor, MyPredictor>(slow);
+                SubsystemManager.RegisterSubsystem(SubsystemKind.CommandPredictor, fast);
+
+                var history = new[] { "hello", "world" };
+                var ids = new HashSet<Guid> { slow.Id, fast.Id };
+
+                CommandPrediction.OnCommandLineAccepted(history);
+                CommandPrediction.OnSuggestionAccepted(slow.Id, "Yeah");
+
+                // The calls to 'EarlyProcessWithHistory' and the feedback methods are queued in thread pool,
+                // so we wait a bit to make sure the calls are done.
+                Thread.Sleep(10);
```

Comment:
Can we get sporadic CI failures? Can we make this more reliable like we do for script tests with Wait-Until()?

Reply:
Update.